packages/connectivity/src/scp-cf/destination/destination.ts

```diff
@@ -324,9 +324,17 @@ const configMapping: Record<string, keyof Destination> = {
 };
 
 export function noDestinationErrorMessage(
-  destination: Destination | DestinationFetchOptions
+  destination: DestinationOrFetchOptionsXOR
 ): string {
   return isDestinationFetchOptions(destination)
     ? `Could not find a destination with name "${destination.destinationName}"! Unable to execute request.`
     : 'Could not find a destination to execute request against and no destination name has been provided (this should never happen)!';
 }
+
+type Without<T> = { [P in keyof T]?: never };
+type XOR<T, U> = (Without<T> & U) | (Without<U> & T);
+
+export type DestinationOrFetchOptionsXOR = XOR<
```

Comment:
Same here, however I think the Xor is not needed, I think a simple `DestinationOrFetchOptions` would suffice.